### PR TITLE
Add auto_pad attribute support to Conv

### PR DESCRIPTION
**Pull request by chinhuang007**
Add auto_pad attribute support to Conv so it will be used along with
pads to determine the padding.

This is to address issue https://github.com/onnx/onnx-tensorflow/issues/502

### CODE REVIEW

**Comment by sdmonov** [comment]
@chinhuang007 the code looks good to me!

**Comment by chinhuang007** [comment]
@tjingrant @fumihwh please review. This would also close https://github.com/onnx/onnx-tensorflow/pull/345 when approved.

File: onnx_tf/handlers/backend/conv_mixin.py [hunk 1]
```diff
@@ -77,73 +107,120 @@ def conv(cls, node, input_dict, transpose=False):
...
-        else:
-          conv_output_shape += [
-              s + pads[i] + pads[spatial_size + i]
-              for i, s in enumerate(output_shape[-2:])
+
+        # calculate output shape
+        if pad == "NOTSET":
+          if output_shape is None:
+            conv_output_shape += [
+                strides[i] * x_spatial_shape[i] +
```
[the remark this answers is not in the document]
Comment (sdmonov) [reply onnx_tf/handlers/backend/conv_mixin.py]: @tjingrant I believe this calculates the output shape for ConvTranspose. The code is branched at https://github.com/onnx/onnx-tensorflow/blob/a44418608210c4662057b10a43e029e87215723b/onnx_tf/handlers/backend/conv_mixin.py#L101
That is why the formula is reverse to the Convolution.

[the remark this answers is not in the document]
Comment (chinhuang007) [reply onnx_tf/handlers/backend/conv_mixin.py]: @tjingrant I changed the variable name to pad_mode to remove the confusion between pad and pads. Please let me know if you still have concerns about the output shade calculation with strides for ConvTranspose or any suggestions to make it clear.

[the remark this answers is not in the document]
Comment (surajkandukuri) [reply onnx_tf/handlers/backend/conv_mixin.py]: @chinhuang007 sorry for my ignorance here, is my issue #502 fixed ? Collaboration in github is also new for me, If the issue is fixed what should i do to have the updated code in my program. if it is still waiting for approval or will take more time what should i see in the github that would say/notify me that my issue has been closed ?

[the remark this answers is not in the document]
Comment (chinhuang007) [reply onnx_tf/handlers/backend/conv_mixin.py]: @surajkandukuri Yes. Issue 502 should be fixed once this PR is approved and merged.

**Pull request merged**